File: VB/HideNavigationItemsExample.Module/Security/CustomSecurityRole.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.Security.Strategy
Imports DevExpress.Persistent.Base
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp.ConditionalAppearance
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.Persistent.BaseImpl.PermissionPolicy

Namespace HideNavigationItemsExample.Module.Security
	<DefaultClassOptions, ImageName("BO_Role"), Appearance("HideHiddenNavigationItemsForAdministrators", AppearanceItemType:="LayoutItem", TargetItems := "HiddenNavigationItems", Visibility := ViewItemVisibility.Hide, Criteria := "IsAdministrative")> _
	Public Class CustomSecurityRole
		Inherits PermissionPolicyRole
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _HiddenNavigationItems As String
		Public Property HiddenNavigationItems() As String
			Get
				Return _HiddenNavigationItems
			End Get
			Set(ByVal value As String)
				SetPropertyValue("HiddenNavigationItems", _HiddenNavigationItems, value)
			End Set
		End Property
		Public Function GetPermissions() As IEnumerable(Of IOperationPermission)
			Dim result As New List(Of IOperationPermission)()
			If (Not String.IsNullOrEmpty(HiddenNavigationItems)) Then
				For Each hiddenNavigationItem As String In HiddenNavigationItems.Split(";"c, ","c)
					result.Add(New NavigationItemPermission(hiddenNavigationItem.Trim()))
				Next hiddenNavigationItem
			End If
			Return result
		End Function
	End Class
End Namespace

File: CS/HideNavigationItemsExample.Module/Security/CustomSecurityRole.cs
using System;
using System.Collections.Generic;
using DevExpress.ExpressApp.Security;
using DevExpress.ExpressApp.Security.Strategy;
using DevExpress.Persistent.Base;
using DevExpress.Xpo;
using DevExpress.ExpressApp.ConditionalAppearance;
using DevExpress.ExpressApp.Editors;

namespace HideNavigationItemsExample.Module.Security {
    [DefaultClassOptions, ImageName("BO_Role")]
    [Appearance("HideHiddenNavigationItemsForAdministrators", AppearanceItemType="LayoutItem", TargetItems = "HiddenNavigationItems", Visibility = ViewItemVisibility.Hide, Criteria = "IsAdministrative")]
    public class CustomSecurityRole : SecuritySystemRole {
        public CustomSecurityRole(Session session) : base(session) { }
        private string _HiddenNavigationItems;
        public string HiddenNavigationItems {
            get { return _HiddenNavigationItems; }
            set { SetPropertyValue("HiddenNavigationItems", ref _HiddenNavigationItems, value); }
        }
        protected override IEnumerable<IOperationPermission> GetPermissionsCore() {
            List<IOperationPermission> result = new List<IOperationPermission>();
            result.AddRange(base.GetPermissionsCore());
            if (!String.IsNullOrEmpty(HiddenNavigationItems)) {
                foreach (string hiddenNavigationItem in HiddenNavigationItems.Split(';', ',')) {
                    result.Add(new NavigationItemPermission(hiddenNavigationItem.Trim()));
                }
            }
            return result;
        }
    }
}
